Frameworks/Foundation/NSIndexPath.mm

```diff
@@ -38,22 +49,28 @@ + (NSIndexPath*)indexPathWithIndexes:(const NSUInteger*)indexes length:(NSUInteg
  @Status Interoperable
 */
 + (NSIndexPath*)indexPathForRow:(NSInteger)row inSection:(NSInteger)section {
-    unsigned indexes[] = { section, row };
-
+    NSUInteger indexes[] = { section, row };
     return [[[self alloc] initWithIndexes:indexes length:2] autorelease];
 }
 
 /**
  @Status Interoperable
 */
-+ (NSIndexPath*)indexPathForItem:(NSInteger)row inSection:(NSInteger)section {
-    NSUInteger indexes[] = { section, row };
++ (NSIndexPath*)indexPathForItem:(NSInteger)item inSection:(NSInteger)section {
+    NSUInteger indexes[] = { section, item };
     return [[[self alloc] initWithIndexes:indexes length:2] autorelease];
 }
 
 /**
  @Status Interoperable
 */
+- (instancetype)init {
```

Comment:
According to documentation, init is the designated initializer, which means all other initializers must call it and this should be the one calling super.  This helps with subclassing.

---
In reply to: [107547799](https://github.com/Microsoft/WinObjC/pull/2279#discussion_r107547799) [](ancestors = 107547799)